default/python/universe_generation/monsters.py

```diff
@@ -1,8 +1,110 @@
+import sys
 import random
 import freeorion as fo
 import util
 import statistics
 import universe_tables
+from galaxy import DisjointSets
+
+
+def populate_monster_fleet(fleet_plan, system):
+    """
+    Create a monster fleet in ''system'' according to ''fleet_plan''
+    """
+
+    # create monster fleet
+    monster_fleet = fo.create_monster_fleet(system)
+    # if fleet creation fails, report an error and try to continue with next system
+    if monster_fleet == fo.invalid_object():
+        errstr = "Python generate_monsters: unable to create new monster fleet %s" % fleet_plan.name()
+        util.report_error(errstr)
+        raise RuntimeError(errstr)
+
+    # add monsters to fleet
+    for design in fleet_plan.ship_designs():
+        # create monster, if creation fails, report an error and try to continue with the next design
```

Comment:
Similar to the comment above, even more so, as in the above case the calling code actually continues with the next system (so the comment isn't strictly wrong, just in the wrong place), but here we don't proceed with the next design, but quit by throwing.